src/main.ts

```diff
@@ -614,6 +614,45 @@ export class Purchases {
         window.addEventListener("popstate", onClose);
       }
 
+      const onFinished = async (
+        operationSessionId: string,
+        redemptionInfo: RedemptionInfo | null,
+      ) => {
+        const event = createCheckoutSessionEndFinishedEvent({
+          redemptionInfo,
+        });
+        this.eventsTracker.trackSDKEvent(event);
+        Logger.debugLog("Purchase finished");
+
+        if (component) {
+          unmount(component);
+        }
+
+        certainHTMLTarget.innerHTML = "";
+        // TODO: Add info about transaction in result.
+        const purchaseResult: PurchaseResult = {
+          customerInfo: await this._getCustomerInfoForUserId(appUserId),
+          redemptionInfo: redemptionInfo,
+          operationSessionId: operationSessionId,
+        };
+        resolve(purchaseResult);
+      };
+
+      const onError = (e: PurchaseFlowError) => {
+        const event = createCheckoutSessionEndErroredEvent({
+          errorCode: e.errorCode?.toString(),
+          errorMessage: e.message,
+        });
+        this.eventsTracker.trackSDKEvent(event);
+
+        if (component) {
```

Comment:
These two (onError and onFinished) are extracted to a function so we can call `unmount(component)` like `onClose`, so that we make sure to return the scroll to the caller in these paths also.